examples/step-5/step-5.cc

```diff
@@ -164,69 +167,75 @@ void Step5<dim>::setup_system()
+void Step5<dim>::assemble_system ()
 {
-  QGauss<dim> quadrature_formula(2);
+  QGauss<dim>  quadrature_formula(2);
 
-  FEValues<dim> fe_values(fe,
-                          quadrature_formula,
-                          update_values | update_gradients |
-                            update_quadrature_points | update_JxW_values);
+  FEValues<dim> fe_values (fe, quadrature_formula,
+                           update_values    |  update_gradients |
+                           update_quadrature_points  |  update_JxW_values);
 
-  const unsigned int dofs_per_cell = fe.dofs_per_cell;
-  const unsigned int n_q_points    = quadrature_formula.size();
+  const unsigned int   dofs_per_cell = fe.dofs_per_cell;
+  const unsigned int   n_q_points    = quadrature_formula.size();
 
-  FullMatrix<double> cell_matrix(dofs_per_cell, dofs_per_cell);
-  Vector<double>     cell_rhs(dofs_per_cell);
+  FullMatrix<double>   cell_matrix (dofs_per_cell, dofs_per_cell);
+  Vector<double>       cell_rhs (dofs_per_cell);
 
-  std::vector<types::global_dof_index> local_dof_indices(dofs_per_cell);
+  std::vector<types::global_dof_index> local_dof_indices (dofs_per_cell);
 
   // Next is the typical loop over all cells to compute local contributions
   // and then to transfer them into the global matrix and vector. The only
   // change in this part, compared to step-4, is that we will use the
   // <code>coefficient</code> function defined above to compute the
   // coefficient value at each quadrature point.
-  for (const auto &cell : dof_handler.active_cell_iterators())
+  for (const auto &cell: dof_handler.active_cell_iterators())
     {
       cell_matrix = 0;
-      cell_rhs    = 0;
+      cell_rhs = 0;
 
-      fe_values.reinit(cell);
+      fe_values.reinit (cell);
 
-      for (unsigned int q_index = 0; q_index < n_q_points; ++q_index)
+      for (unsigned int q_index=0; q_index<n_q_points; ++q_index)
         {
-          const double current_coefficient =
-            coefficient<dim>(fe_values.quadrature_point(q_index));
-          for (unsigned int i = 0; i < dofs_per_cell; ++i)
+          const double current_coefficient = coefficient<dim>
+                                             (fe_values.quadrature_point (q_index));
+          for (unsigned int i=0; i<dofs_per_cell; ++i)
             {
-              for (unsigned int j = 0; j < dofs_per_cell; ++j)
-                cell_matrix(i, j) +=
-                  (current_coefficient * fe_values.shape_grad(i, q_index) *
-                   fe_values.shape_grad(j, q_index) * fe_values.JxW(q_index));
-
-              cell_rhs(i) += (fe_values.shape_value(i, q_index) * 1.0 *
+              for (unsigned int j=0; j<dofs_per_cell; ++j)
+                cell_matrix(i,j) += (current_coefficient *
+                                     fe_values.shape_grad(i,q_index) *
+                                     fe_values.shape_grad(j,q_index) *
+                                     fe_values.JxW(q_index));
```

Comment:
About the only places that I can see and where I absolutely agree with @bangerth that they are reindented suboptimally are these long mathematical expressions of bilinear forms (well and some awkward operator output). As I said, we can instruct clang format to not apply line breaks here. There are two possibilities to achieve this:
```
cell_matrix(i, j) += (current_coefficient *              //
                      fe_values.shape_grad(i, q_index) * //
                      fe_values.shape_grad(j, q_index) * //
                      fe_values.JxW(q_index));           //
```
```
/* clang-format off */
cell_matrix(i, j) += (current_coefficient *
                      fe_values.shape_grad(i, q_index) *
                      fe_values.shape_grad(j, q_index) *
                      fe_values.JxW(q_index));
/* clang-format on */
```
I actually like the first variant. There, clang format will still reindent everything but simply keeps the line breaks.

Reply:
I have commented more about this over here https://github.com/tamiko/dealii/pull/10#discussion_r191663138
I think that the first variant is nice as well. I'm a bit undecided whether we would like to remove those trailing `//` characters (or `clang-format on/off`) as part of the process generating the online documentation, but in the end I'm fine to accept the variant that the majority agrees upon.

Reply:
We could use the trailing slashes to explain again what each part of the formula means. For everything else, remember that people eventually look at the code -- they will ask themselves and the instructor what exactly any of our markup means. We could lessen the impact by processing the source files in some way before we install them, but that's suboptimal. I think that leaving some macro in is not a good option.

If we want to go with any kind of mark-up, there are two questions:
* Who is going to do the work? We have ~60 tutorial programs, so someone will have to find 100-150 places that need to be fixed. That'll take several hours.
* What this suggestion really is is to work around the limitations of a tool. That's annoying -- do we really want to commit to this?

Reply:
@bangerth We will have to go through all examples in the next weeks anyway. This includes
 - necessary changes and documentation updates for `AffineConstraints` (once merged)
 - C++11 improvements

We could add markup fixes to the list. With 4 people working on it it is also only 15 example steps per person. (Preferably one pull request per example step and a list to tick off) That's doable over a course of a month. (This implies we trust each other to a certain extent with respect to the markup changes.)